File: adventofcode/06/6.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    string line;
    int sol1 = 0, sol2 = 0, c = 0;
    map<char, int> y;
    while(getline(cin, line)){
        if (!line.empty()) {
            c++;
            for (int i = 0; i < line.size(); i++) y[line[i]]++;
        } else {
            sol1 += y.size();
            for (map<char, int>::iterator i = y.begin(); i != y.end(); i++) if (i->second == c) sol2++;
            c = 0;
            y = map<char, int>();
        }       
    }
    cout << "Problem1: " << sol1 << endl << "Problem2: " << sol2 << endl;
}

File: adventofcode/06/6.py
from sys import stdin

sol1 = sol2 = 0
s1, s2 = [set(), set({c for c in 'abcdefghijklmnopqrstuvwxyz'})]
for line in stdin:
    if line.rstrip() == '':
        sol1 += len(s1)
        sol2 += len(s2)
        s1, s2 = [set(), set({c for c in 'abcdefghijklmnopqrstuvwxyz'})]
    else:
        s1=s1.union(set([c for c in line.rstrip()]))
        s2=s2.intersection(set([c for c in line.rstrip()]))
print("Problem1: {}\nProblem2: {}".format(sol1, sol2))